controller/login.go

```diff
@@ -112,6 +113,47 @@ func (c *LoginController) Authorize(ctx *app.AuthorizeLoginContext) error {
 	return c.auth.Perform(ctx, oauth, brokerEndpoint, entitlementEndpoint, profileEndpoint, whitelist, c.configuration.GetAuthNotApprovedRedirect())
 }
 
+// getEntitlementResourceRequestPayload creates the object which would have the information about which spaces/resources
+// the entitlements' info would need to be fetched for.
+
+func (c *LoginController) getEntitlementResourceRequestPayload(ctx context.Context, token *string) (*auth.EntitlementResource, error) {
+	loggedInIdentityID, err := c.tokenManager.Extract(*token)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"err": err,
+		}, "Unable to get ID from access token")
+		return nil, errors.NewInternalError(ctx, errs.Wrap(err, "Unable to get ID from access token"))
+	}
+
+	// get the user object as well for this identity
+	queryResult, err := c.identities.Query(account.IdentityFilterByID(loggedInIdentityID.ID), account.IdentityWithUser())
+	if err != nil || len(queryResult) == 0 {
+		log.Error(ctx, map[string]interface{}{
+			"err":         err,
+			"identity_id": *loggedInIdentityID,
+		}, "Unable to query Identity")
+		return nil, errors.NewInternalError(ctx, errs.Wrap(err, "Unable to query Identity"))
+	}
+	loggedInIdentity := queryResult[0]
+	contextInfoLoggedInIdentity := loggedInIdentity.User.ContextInformation
+	if contextInfoLoggedInIdentity == nil {
+		return nil, nil
+	}
+
+	var spacesToGetEntitlementsFor []auth.ResourceSet
+	for _, v := range contextInfoLoggedInIdentity["recentSpaces"].([]interface{}) {
+		recentSpaceID := v.(string)
```

Comment:
Please, do this kind of casting. They can avoid us throwing a panic in some scenerios `recentSpaceID, ok := v.(string)`